### ISSUE TITLE
how to set weight decay of BN weights to zero? 

### ISSUE THREAD

**Issue by pluto16** [marvis/pytorch-yolo2#34]
Hi ,In your release note, you write 
"Fix a lot of problems
weight decay on bn weights and bias weight should zero..."
and in train.py, from line 98 to line 105
"
params_dict = dict(model.named_parameters())
params = []
for key, value in params_dict.items():
    if key.find('.bn') >= 0 or key.find('.bias') >= 0:
        params += [{'params': [value], 'weight_decay': 0.0}]
    else:
        params += [{'params': [value], 'weight_decay': decay*batch_size}]
optimizer = optim.SGD(model.parameters(), lr=learning_rate/batch_size, momentum=momentum, dampening=0, weight_decay=decay*batch_size)
"
params seemed not be used.

**Comment by marvis** [comment]
You are right! weight decay is not the problem.

**Comment by pluto16** [comment]
Hi, Would you mind tell me how long you have trained on pascal voc 2007+2012 train+val , and what kind of gpu you have used.

**Comment by marvis** [comment]
4 titanx, about 14hous, get Mean AP = 0.7270

**Comment by wjuny** [comment]
@marvis Did you use same parameters setting(learning rate,batch size,data augmentation and loss function) in this repository to get 0.7270 mAP?
